domain/src/main/java/org/oppia/domain/oppialogger/analytics/AnalyticsController.kt

```diff
@@ -154,4 +173,79 @@ class AnalyticsController @Inject constructor(
       )
       .build()
   }
+
+  /**
+   * Checks network connectivity of the device.
+   * Saves the [eventLog] to the [eventLogStore] in the absence of it.
+   * Uploads to remote service in the presence of it.
+   */
+  private fun checkNetworkAndManageEventLog(eventLog: EventLog) {
+    when (networkConnectionUtil.getCurrentConnectionStatus()) {
+      NONE -> addEventLog(eventLog)
+      else -> eventLogger.logEvent(eventLog)
+    }
+  }
+
+  /**
+   * Adds an event to the storage.
+   * At first, it checks if the size of the store isn't exceeding [eventLogStorageCacheSize]
+   * If the limit is exceeded then the least recent event is removed from the [eventLogStore]
+   * After this, the [eventLog] is added to the store.
+   * */
+  private fun addEventLog(eventLog: EventLog) {
```

Comment:
Perhaps: ``cacheEventLog`` or ``saveEventLog``. 'add' is a bit ambiguous; we should be specific where the event log is going to provide context at the callsite.

Reply:
Done.